src/main/java/org/broadinstitute/hellbender/tools/funcotator/FilterFuncotations.java

```diff
@@ -0,0 +1,201 @@
+
+    static final String ONE_LINE_SUMMARY = "Filter variants based on clinically-significant Funcotations.";
+    static final String SUMMARY = ONE_LINE_SUMMARY +
+            "\nThis proof-of-concept tool is an example for how to parse and use the VCF output of Funcotator." +
+            "\nCurrently hard-coded to look for specific Funcotations from:" +
+            "\n  * ClinVar (http://www.clinvar.com/)" +
+            "\n  * Exome Aggregation Consortium (ExAC) (http://exac.broadinstitute.org/)" +
+            "\n  * Laboratory for Molecular Medicine (LMM) (http://personalizedmedicine.partners.org/laboratory-for-molecular-medicine/)";
+
+    /**
+     * The version of the Human Genome reference which was used when Funcotating the input VCF.
+     *
+     * Used to derive names of Gencode Funcotations.
+     */
+    public enum ReferenceVersion {
+        hg19(19), hg38(27);
+
+        private final int gencodeVersion;
+
+        ReferenceVersion(int gencodeVersion) {
+            this.gencodeVersion = gencodeVersion;
+        }
+
+        public int getGencodeVersion() {
+            return gencodeVersion;
+        }
+    }
+
+    @Argument(
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            doc = "Output VCF file to which filtered variants should be written.")
+    protected File outputFile;
+
+    @Argument(
+            fullName =  FuncotatorArgumentDefinitions.REFERENCE_VERSION_LONG_NAME,
+            doc = "The version of the Human Genome reference which was used to Funcotate the input VCF."
+    )
+    protected ReferenceVersion referenceVersion;
+
+    private VariantContextWriter outputVcfWriter;
+    private String[] funcotationKeys;
+    private final List<FuncotationFilter> funcotationFilters = new ArrayList<>();
+
+    @Override
+    public void onTraversalStart() {
+        registerFilters();
+        final VCFHeader vcfHeader = getHeaderForVariants();
+
+        final VCFInfoHeaderLine funcotationHeaderLine = vcfHeader.getInfoHeaderLine(VcfOutputRenderer.FUNCOTATOR_VCF_FIELD_NAME);
+        if (funcotationHeaderLine != null) {
+            funcotationKeys = FuncotatorUtils.extractFuncotatorKeysFromHeaderDescription(funcotationHeaderLine.getDescription());
+            outputVcfWriter = createVCFWriter(outputFile);
+            vcfHeader.addMetaDataLine(new VCFFilterHeaderLine(FilterFuncotationsConstants.NOT_CLINSIG_FILTER,
+                    FilterFuncotationsConstants.NOT_CLINSIG_FILTER_DESCRIPTION));
+            vcfHeader.addMetaDataLine(new VCFInfoHeaderLine(FilterFuncotationsConstants.CLINSIG_INFO_KEY, 1,
+                    VCFHeaderLineType.String, FilterFuncotationsConstants.CLINSIG_INFO_KEY_DESCRIPTION));
+            outputVcfWriter.writeHeader(vcfHeader);
+        } else {
+            throw new UserException.BadInput("Input VCF does not have Funcotator annotations.");
```

Comment:
Can you mention in this exception that it's an error in the header as opposed to the data itself?